### ISSUE TITLE
Different results from events file and tensorboard

### ISSUE THREAD

**Issue by ErinYang96** [Kamlesh364/Mask-RCNN-TF2.7.0-keras2.7.0#11]
Hi,

I am having some problems in reading my training results.
When I train the head layers for 80 epochs, and fine tune all layers for 160 epochs, I can only read the 80 epochs from the events files. However, when I read the learning curve by tensorboard, I can see changes in all 160 epochs. Could you briefly explain why this happens? 
This is the results from events file.
![image](https://user-images.githubusercontent.com/100861190/176697456-ac057c4a-0873-4f61-ab53-735f8d30f3f0.png)
This is the results from tensorboard (3 curves for training dataset as examples)
![image](https://user-images.githubusercontent.com/100861190/176697675-40d42ac8-c96c-4a12-b81b-d9d50e41641a.png)
![image](https://user-images.githubusercontent.com/100861190/176697716-319a42d9-8969-4e63-9e79-c4afda3d07e6.png)
![image](https://user-images.githubusercontent.com/100861190/176697761-9f0ad477-2a3f-4b03-b9ee-93b422e567c3.png)
It seems to me that the two results are quite different. I am looking forward to your explanation!
Many thanks for your kindness in advance.

Best,
Erin

**Comment by ErinYang96** [comment]
Sorry for my mistake in reading the events file. It turns out that the log folder contains two event file. After switching to another file, I get the training result is as follow.
![trout20220628T1753_sum](https://user-images.githubusercontent.com/100861190/176708972-33e5a1f0-c6d8-4a77-92f7-18170e1e9e40.jpg)
The results seem fine, only with the mrcnn_class_loss keeps fluctuating (By the way, I have only one class defined in this model). Do you have any idea for the reason? 
In addition, there are only 80 epochs in the events file, while tensorboard can read as far as 160 epochs. Is this repo designed to do so?
Thanks for your patience again!

**Comment by Kamlesh364** [comment]
@ErinYang96 
I would suggest you not to go for 80 epochs to train the head-layers. It would be better if you keep it at-most 5 epochs for "head" and rest for "all" layers.
do share your experience with suggested changes to understand the issue better.

regards,
Kamlesh

**Comment by ErinYang96** [comment]
@Kamlesh364 I have found out that the event files produced by this Mask_RCNN model record loss functions for first and second stages seperately. I try to combine the two events files together, and the result is exactly the same as tensorboard has shown. Thanks for your kind advice, I will try and let you know how it works.

**Comment by ErinYang96** [comment]
@Kamlesh364 I have got the new learning curves following your advice (see below). And the mrcnn_class_loss() finally gets decreased thanks to your guidance!
![image](https://user-images.githubusercontent.com/100861190/177396138-86458b05-e531-4754-b067-3079b0f6d504.png)
However, there is a relatively large gap between training and validation loss. I think there may be two reasons: 

1. the training dataset is relatively small compared with the validation dataset;
2. the validation dataset is not representative enough to evaluate the model.

Could you give me some further advice on how I should improve my model performace? 
Note that for this training, I train the head layer for 5 epochs, and fine tune all layers for 120 epochs. I have 253 images in training dataset and 187 images for validation dataset. If you need more information to decide how to improve the model, please don't hesitate to comment below.
I am looking forward to your response!

**Comment by Kamlesh364** [comment]
@ErinYang96 
To increase the performance. I would suggest you to reduce the validation dataset and increase the training dataset.
Specifically, if you have total 440(253+187) images, try-
1. 95%-Training, 5%-Validation
2. 90%-Training, 10%-Validation
3. 85%-Tarining, 15%- Val.
4. 80%-Training, 20%-Val.

Try above combinations and see which one gives you best results. Please do let us know which one gave you what results 

best regards,
Kamlesh

**Comment by ErinYang96** [comment]
@Kamlesh364 
I have restarted my training with 362 images for training datasets (85%) and 72 images (15%) for validation datasets, with 5 epochs for head layers and 120 epochs for the all layers.
The results are improving again, see below.
![Tensorboard](https://user-images.githubusercontent.com/100861190/178114704-9a4dfa97-f108-4e0b-9ab4-4b3f9b78058a.jpg)
I haven't tested the rest combinations that you listed yet, since I suppose there will not be much difference in the magnitude level. I am thinking about add more images to my datasets or improve my pre-processing of images. 
If I have any great improvements, I will update my results and methods here. Of course, any suggestions from you are always welcomed, since they are always so useful!
Thanks for spending in helping me improving my code again!

**Comment by Kamlesh364** [comment]
@ErinYang96 
I am happy to hear that my suggestions helped you improving the performance of your model.

thank you for sharing your results.

best regards,
Kamlesh
